open_lm/main.py

```diff
@@ -636,14 +668,27 @@ def main(args):
         else:
             total_steps = (data["train"].dataloader.num_batches) * args.epochs
 
-        scheduler = cosine_lr(
-            optimizer,
-            args.lr,
-            args.warmup,
-            total_steps,
-            args.lr_cooldown_end,
-            args.force_min_lr,
-        )
+        if args.lr_scheduler == "cosine":
+            scheduler = cosine_lr(
+                optimizer,
+                args.lr,
+                args.warmup,
+                total_steps,
+                args.lr_cooldown_end,
+                args.force_min_lr,
+            )
+        elif args.lr_scheduler == "const":
+            scheduler = const_lr(
+                optimizer,
+                args.lr,
+                args.warmup,
+                # total_steps,
+                # args.lr_cooldown_end,
+                # args.force_min_lr,
+            )
+        else:
+            logging.error(f"Unknown scheduler, {args.lr_scheduler}. Available options are: cosine, const.")
```

Comment:
Can you make this raise a ValueError exception instead?

Reply:
sure no problem